axum/src/extract/form.rs

```diff
@@ -60,7 +60,7 @@ where
                 .map_err(FailedToDeserializeQueryString::new::<T, _>)?;
             Ok(Form(value))
         } else {
-            if !has_content_type(req, "application/x-www-form-urlencoded")? {
+            if !has_content_type(req, mime::WWW_FORM_URLENCODED.as_ref())? {
```

Comment:
One small thing though, we can probably update this helper function to take `&Mime`, right?

Reply:
I thought about it for a while, and felt that change this to take `&Mime` is not necessary now, because we still need to call `mime.as_ref()` in `has_content_type`:
```
pub(crate) fn has_content_type<B>(
    req: &RequestParts<B>,
    expected_content_type: &mime::Mime,
) -> Result<bool, HeadersAlreadyExtracted> {
    ...
    Ok(content_type.starts_with(expected_content_type.as_ref()))
}
```

This helper function can take `&str` now, which is much more flexible than can only take `&mime::Mime` which also has no gain.

What do you think?

Reply:
I think it will actually be useful to restrict the function like that because it will enforce (more) consistent usage of the constants from the mime crate.

Reply:
That's right, I will do that.